### DISCUSSION TITLE
v2.2.0

### DISCUSSION THREAD

**Discussion by hyrious** [netless-io/flat#1886] [Release]
### Features

* **flat-components**: add avatar window ([#1847](https://github.com/netless-io/flat/issues/1847)) ([54fbc8f5](https://github.com/netless-io/flat/commit/54fbc8f5d30197b5dd1f1359bbe12d9c3192b4d1))

## Commit History
* refactor(flat-pages): open users panel on click users count by @hyrious in https://github.com/netless-io/flat/pull/1838
* fix(desktop): nav avatar should only fail when src exists by @hyrious in https://github.com/netless-io/flat/pull/1839
* fix(desktop): dedupe flat-server-api and flat-stores by @hyrious in https://github.com/netless-io/flat/pull/1840
* fix(i18n): fix banned text by @hyrious in https://github.com/netless-io/flat/pull/1841
* feat(flat-components): add avatar window by @hyrious in https://github.com/netless-io/flat/pull/1847
* refactor(flat-components): change model by @hyrious in https://github.com/netless-io/flat/pull/1849
* fix(service-providers): remove room writable hack by @hyrious in https://github.com/netless-io/flat/pull/1850
* refactor(flat-stores): delete avatar window on leaving stage by @hyrious in https://github.com/netless-io/flat/pull/1851
* refactor(flat-pages): hide toolbar on avatar grid mode by @hyrious in https://github.com/netless-io/flat/pull/1852
* chore(library): upgrade @netless/app-slide by @hyrious in https://github.com/netless-io/flat/pull/1854
* refactor(flat-pages): implement grid layout (up to 6 items) by @hyrious in https://github.com/netless-io/flat/pull/1853
* refactor(flat-pages): show join options in more places by @hyrious in https://github.com/netless-io/flat/pull/1855
* fix(flat-stores): joiner off stage leaves window by @hyrious in https://github.com/netless-io/flat/pull/1856
* chore(desktop): ignore environment variables with spaces in names by @hyrious in https://github.com/netless-io/flat/pull/1859
* refactor(flat-stores): user windows treat empty array as normal mode by @hyrious in https://github.com/netless-io/flat/pull/1861
* fix(library): duplicate app-slide by @hyrious in https://github.com/netless-io/flat/pull/1862
* fix(library): upgrade app countdown by @hyrious in https://github.com/netless-io/flat/pull/1863
* refactor(service-providers): use 384x216 (16:9) source video by @hyrious in https://github.com/netless-io/flat/pull/1864
* refactor(flat-pages): add gap to one-to-one avatars by @hyrious in https://github.com/netless-io/flat/pull/1865
* refactor(desktop): update macos app icons by @hyrious in https://github.com/netless-io/flat/pull/1866
* refactor(flat-stores): automatically on stage in small class mode by @hyrious in https://github.com/netless-io/flat/pull/1867
* refactor(desktop): load RTC at the very beginning by @hyrious in https://github.com/netless-io/flat/pull/1869
* chore(desktop): update mac icon format and size by @vince-hz in https://github.com/netless-io/flat/pull/1868
* chore(desktop): fix terminal launch script by @hyrious in https://github.com/netless-io/flat/pull/1871
* chore(desktop): add "mas" build config (mac app store) by @vince-hz in https://github.com/netless-io/flat/pull/1872
* refactor(flat-pages): recalculate grid layout to keep window ratio by @hyrious in https://github.com/netless-io/flat/pull/1875
* refactor(service-providers): share screen with audio on web by @hyrious in https://github.com/netless-io/flat/pull/1874
* fix(service-providers): use standalone avatar for testing devices by @hyrious in https://github.com/netless-io/flat/pull/1876
* refactor(flat-stores): auto on stage in small class only for once by @hyrious in https://github.com/netless-io/flat/pull/1877
* chore(web): add google analytics to web dev site by @hyrious in https://github.com/netless-io/flat/pull/1878
* refactor(flat-components): show avatar windows that has left by @hyrious in https://github.com/netless-io/flat/pull/1879
* refactor(service-providers): enable audio with share screen in electron by @hyrious in https://github.com/netless-io/flat/pull/1881
* chore(project): add google analytics to web and electron by @hyrious in https://github.com/netless-io/flat/pull/1882
* fix(desktop): incorrect origin in development build by @hyrious in https://github.com/netless-io/flat/pull/1883
* fix(service-providers): fix electron share screen with audio by @hyrious in https://github.com/netless-io/flat/pull/1884
* chore(version): upgrade version to v2.2.0 by @hyrious in https://github.com/netless-io/flat/pull/1885

## New Contributors
* @vince-hz made their first contribution in https://github.com/netless-io/flat/pull/1868

**Full Changelog**: https://github.com/netless-io/flat/compare/v2.1.0...v2.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/netless-io/flat/releases/tag/v2.2.0'>v2.2.0</a>.</em>

**Comment by Leooeloel** [comment]
Flat V2.2.0 new version effect demo：[https://i.imgur.com/HKbX2Wo.mp4](https://i.imgur.com/HKbX2Wo.mp4?fbclid=IwAR0fRnfCiKCuLbsiZuSsKlHsnNUesXMAThrK4ZnJCe33rU-k2Ys0qWTyK8Y)
